src/OrleansProviders/Streams/Memory/MemoryBatchContainer.cs

```diff
@@ -9,10 +7,10 @@
 
 namespace Orleans.Providers
 {
-    [Serializable]
     internal class MemoryBatchContainer<TSerializer> : IBatchContainer
```

Comment:
Julian explained what's going on here.
This solution is -ok- but imo worse than the previous approach.
The problem is that all IBatchContainers, and all interfaces included in the stream adapter pattern, are application extension points.  They are how developers write new persistent stream providers.  In the previous approach, we had to change quite a few internal classes to get the serialization manager to the batch container, but those were all our code.  In this pattern we don't have to do that (hence less code changes) but the application layer code needs to be more complicated as it needs to figure out some trick to getting the serialization manager, like we've done in DefaultMemoryMessageBodySerializer.  We've not only made the serialization manager no longer global, we've made it the application layer's problem to figure out how to get it, instead of passing it to systems that need it.